### PR TITLE
refactor: redis bridge to its own application

### PR DESCRIPTION
**Pull request by kjellwinblad**
This PR refactor the redis bridge to its own application. The open source redis connector is also refactored to its own application.

Fixes:
https://emqx.atlassian.net/browse/EMQX-9534

### <samp>🤖 Generated by Copilot at 94bb0e3</samp>

This pull request introduces a new application `emqx_redis` that provides a common module for managing Redis connections and operations. It also refactors and updates the existing applications `emqx_authn`, `emqx_authz`, and `emqx_bridge_redis` to use the `emqx_redis` module instead of the deprecated `emqx_connector_redis` module. Additionally, it renames some modules and files to simplify and unify the naming conventions. Finally, it adds or updates the license, readme, and test files for the affected applications.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [] Added tests for the changes
- [] Changed lines covered in coverage report
- [x] Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [x] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [] Schema changes are backward compatible

### CODE REVIEW

**Comment by kjellwinblad** [comment]
PR for fixing the i18 repo:

https://github.com/emqx/emqx-i18n/pull/4

File: apps/emqx_redis/src/emqx_redis.app.src [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{application, emqx_redis, [
+    {description, "EMQX Enterprise Oracle Database Connector"},
```
Comment (paulozulato) [inline apps/emqx_redis/src/emqx_redis.app.src:2]: EMQX Enterprise Redis Database Connector"

File: mix.exs [hunk 2]
```diff
@@ -412,6 +412,7 @@ defmodule EMQXUmbrella.MixProject do
           emqx_bridge_timescale: :permanent,
           emqx_bridge_sqlserver: :permanent,
           emqx_oracle: :permanent,
+          emqx_redis: :permanent,
```
Comment (thalesmg) [inline mix.exs:415]: I think we need to add the `emqx_bridge_redis` as well (and in rebar.config) to include it in the release.

File: lib-ee/emqx_ee_connector/rebar.config [hunk 3]
```diff
@@ -3,7 +3,8 @@
 {deps, [
   {hstreamdb_erl, {git, "https://github.com/hstreamdb/hstreamdb_erl.git", {tag, "0.2.5"}}},
   {emqx, {path, "../../apps/emqx"}},
-  {emqx_utils, {path, "../../apps/emqx_utils"}}
+  {emqx_utils, {path, "../../apps/emqx_utils"}},
+  {emqx_redis, {path, "../../apps/emqx_redis"}}
```
Comment (thalesmg) [inline lib-ee/emqx_ee_connector/rebar.config:7]: Q: do we need it here now that it was moved to its own app?

File: mix.exs (hunk 2 above)
Comment (kjellwinblad) [reply mix.exs:415]: It is already added as there was and app template for redis before this PR. See higher up in the list.

File: lib-ee/emqx_ee_connector/rebar.config (hunk 3 above)
Comment (kjellwinblad) [reply lib-ee/emqx_ee_connector/rebar.config:7]: You are not needed. I did the refactoring in two steps (broke out the open source connector first) and it seems like I forgot to delete this when it was not needed any longer.

Comment (kjellwinblad) [reply lib-ee/emqx_ee_connector/rebar.config]: Fixed in [5de6aed](https://github.com/emqx/emqx/pull/11139/commits/5de6aedc31956b49cc5af1dc5c6f273b8e55b94b)

File: apps/emqx_redis/src/emqx_redis.app.src (hunk 1 above)
Comment (kjellwinblad) [reply apps/emqx_redis/src/emqx_redis.app.src]: Thanks! Fixed in [5de6aed](https://github.com/emqx/emqx/pull/11139/commits/5de6aedc31956b49cc5af1dc5c6f273b8e55b94b)

**Review by thalesmg — APPROVED** [review]

**Review by paulozulato — APPROVED** [review]

**Pull request merged**